pkg/command/feature.go

```diff
@@ -0,0 +1,57 @@
+package command
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/afero"
+	"github.com/xenitab/gitops-promotion/pkg/config"
+	"github.com/xenitab/gitops-promotion/pkg/git"
+	"github.com/xenitab/gitops-promotion/pkg/manifest"
+)
+
+// FeatureCommand is similar to NewCommand but creates a PR with a temporary deployment of the application.
+// A totally new application will be created instead of overriding the existing application deployment.
+func FeatureCommand(ctx context.Context, cfg config.Config, repo *git.Repository, group, app, tag, feature string) (string, error) {
+	state := git.PRState{
+		Env:     cfg.Environments[0].Name,
+		Group:   group,
+		App:     app,
+		Tag:     tag,
+		Sha:     "",
+		Feature: feature,
+		Type:    git.PRTypeFeature,
+	}
+	featureLabelSelector, err := cfg.GetFeatureLabelSelector(state.Group, app)
+	if err != nil {
+		return "", fmt.Errorf("feature deployment does not work without configuring a feature label selector: %w", err)
+	}
+	fs := afero.NewBasePathFs(afero.NewOsFs(), repo.GetRootDir())
+	err = manifest.DuplicateApplication(fs, state, featureLabelSelector)
```

Comment:
I would have preferred to pass in the root dir and use afero from within DuplicateApplication so as to reduce the usage of a lib that may not be used elsewhere.

Reply:
The plan is to use afero in the whole application. Doing it this way is a common pattern in a lot of go applications, especially because it makes testing a lot simpler.